src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_utils.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2020-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "jit_brgemm_utils.hpp"
+#include "emitters/utils.hpp"
+#include "utils/general_utils.h"
+
+using namespace Xbyak;
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu::x64;
+
+namespace ov {
+namespace intel_cpu {
+namespace jit_brgemm_utils {
+
+cpu_isa_t get_primitive_isa(const ov::element::Type& dt_in0, bool is_with_amx) {
+    auto isa = isa_undef;
+#define SUPPORT(X, Y) if (mayiuse(X)) { isa = X; } else { Y }
+#define SUPPORT_ONE(X, MESSAGE) SUPPORT(X, OV_CPU_JIT_EMITTER_THROW(MESSAGE);)
+#define SUPPORT_TWO(X, Y, MESSAGE) SUPPORT(X, SUPPORT_ONE(Y, MESSAGE))
+
+    // Note: AMX might be not used even if it's supported by the hardware, check the BrgemmToBrgemmCPU pass for details
+    if (is_with_amx) {
+        SUPPORT_ONE(avx512_core_amx, "Unsupported hardware configuration: amx is supported only on avx512 platforms")
+    } else if (dt_in0 == ov::element::bf16) {
+        SUPPORT_ONE(avx512_core_bf16, "Unsupported hardware configuration: bf16 is supported only on avx512 platforms")
+    } else if (one_of(dt_in0, ov::element::u8, ov::element::i8)) {
+        SUPPORT_TWO(avx512_core_vnni, avx2_vnni, "Unsupported hardware configuration: int8 is supported only on vnni platforms")
+    } else {
+        SUPPORT_TWO(avx512_core, cpu::x64::avx2, "Unsupported hardware configuration: brgemm requires at least avx2 isa")
+    }
+    return isa;
+#undef SUPPORT_TWO
+#undef SUPPORT_ONE
+#undef _SUPPORT
+}
+
+BRGEMM_TYPE get_brgemm_type(const ov::element::Type& element_type_a, const Dimension& K_dim, const Dimension& N_dim) {
+    if (element_type_a == element::f32)
+        return BRGEMM_TYPE::STANDARD;
+    const auto brgemmVNNIFactor = 4 / element_type_a.size();
+    if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx) &&
+        K_dim.is_static() && K_dim.get_length() % brgemmVNNIFactor == 0 &&
+        N_dim.is_static() && N_dim.get_length() % brgemmVNNIFactor == 0)
+        return BRGEMM_TYPE::WITH_AMX;
+    // Note: this condition reproduces logic from the OneDNN Brgemm implementation. This is needed to align with the
+    // backend requirements. More details in onednn/src/cpu/x64/brgemm/brgemm_utils.cpp
+    if (element_type_a == ov::element::i8 &&
+       !dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx2_vnni_2))
+       return BRGEMM_TYPE::WITH_COMPENSATIONS;
+    OV_CPU_JIT_EMITTER_THROW("Failed to determine brgemm mode");
+}
```

Comment:
Probably I missed something (at the moment there is evening on Friday 😅), but how we should handle the case on non-amx systems with `element_type_a = u8` or `element_type_a = bf16`? 🤔 
I just compare with [header](https://github.com/openvinotoolkit/openvino/blob/9fafcabb7c7b221821b0b77616d9967c8d13223f/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/brgemm_to_brgemm_cpu.hpp) of the corresponding transformation. Moreover, can we reuse the existing enum `BrgemmCopyB::Type`? (but to move to `brgemm` section and rename, for sure)

Reply:
> element_type_a = u8 or element_type_a = bf16
Fixed, thanks. That's exactly the point why all this logic must be in one place: it's easy to mess it up

>Moreover, can we reuse the existing enum BrgemmCopyB::Type? (but to move to brgemm section and rename, for sure)
Didn't get it. Where do you want to reuse BrgemmCopyB::Type? As a return type of `get_brgemm_type`?

Reply:
> I think we can do the opposite: reuse this enum from the utils in the Brgemm and BrgemmCopyB nodes (just store the result of get_brgemm_type for example).

Or to make so at least. Just to avoid different `enum` structures which should be fully the same. If we need to change one of them (for example, enum in `brgemm_utils`), we have to update the corresponding enum in `BrgemmCPU`. I'm not sure that it's good dependency. Just thinking out loud, no more comments from my side 😃

Reply:
Decided to do the refactoring, please take a look at the last commit